csharp-basic-topics/AwaitInForEachLoops/AwaitInForEachLoops/TaskWhenAll.cs

```diff
@@ -0,0 +1,35 @@
+﻿
+
+namespace AwaitInForEachLoop
+{
+    public class TaskWhenAllInLoop
+    {
+        public static async Task<List<Task>> ResultAsync()
+        {
+            var numbers = new List<int> { 10, 20, 30, 40, 50, 60, 70, 80, 90 };
+
+            var processingTasks = new List<Task>();
+
+            foreach (int number in numbers)
+            {
+            
+                processingTasks.Add(ProcessNumberAsync(number));
+            }
+
+            await Task.WhenAll(processingTasks);
+
+            Console.WriteLine("Done Processing");
+
+            return processingTasks;
+        }
+
+        public static async Task ProcessNumberAsync(int number)
+        {
+            // Simulate an asynchronous operation
```

Comment:
Here also print the processing line, and add the number to the result list once it is processed.

Reply:
i already added the number to the result list
var result = new List<Task>();

foreach (int number in numbers)
{            
    result.Add(ProcessNumberAsync(number));
}

Reply:
FYI, we are not adding the number in the result list as we are doing for another code example. Here we are adding the task to the result list. I was expecting the result to be the list of numbers in the order it is processed to make the example consistent with the other one. For now, I will accept this solution but it may come back from SE review.